### PR TITLE
refactor: reply error for commands

### PR DESCRIPTION
**Pull request by danielvladco**

### CODE REVIEW

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go [hunk 1]
```diff
@@ -101,18 +101,36 @@ type apiInfoProvider interface {
...
+	SendError(err error)
+	RcvError() error
+}
+
+type ErrorChan chan error
+
+func NewErrorChan() ErrorChan           { return make(chan error, 1) }
+func (e ErrorChan) SendError(err error) { e <- err }
+func (e ErrorChan) Close()              { close(e) }
+func (e ErrorChan) RcvError() error     { return <-e }
```
Comment (aduminuc) [inline backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:117]: Is this blocking if there is no error?

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go [hunk 2]
```diff
@@ -189,9 +219,13 @@ func (l *learnAndDetectBFLA) run(ctx context.Context) {
...
+		case command, ok := <-l.commandsCh:
 			if ok {
-				if err := l.commandsRunner(ctx, feedback); err != nil {
+				err := l.commandsRunner(ctx, command)
+				if cmdErr, ok := command.(CommandWithError); ok {
+					cmdErr.SendError(err)
+				}
+				if err != nil {
 					l.errCh <- err
 				}
```
Comment (aduminuc) [inline backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:230]: What is it for?

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go [hunk 3]
```diff
@@ -189,9 +219,13 @@ func (l *learnAndDetectBFLA) run(ctx context.Context) {
 
 	for {
 		select {
-		case feedback, ok := <-l.commandsCh:
+		case command, ok := <-l.commandsCh:
 			if ok {
-				if err := l.commandsRunner(ctx, feedback); err != nil {
+				err := l.commandsRunner(ctx, command)
+				if cmdErr, ok := command.(CommandWithError); ok {
+					cmdErr.SendError(err)
```
[the remark this answers is not in the document]
Comment (danielvladco) [reply backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:226]: the logic is: if the command is of type 'WithError' wee make sure to send the error back to the channel where the handler will be waiting for the error to come back

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go (hunk 2 above)
Comment (danielvladco) [reply backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:230]: this is an error channel for handling errors, the idea is if we ever want to have a more complex error handling logic we can just  send the errors and the error handling component can be independently modified to handle any logic. But right now it just logs the errors :)

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go (hunk 1 above)
Comment (danielvladco) [reply backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:117]: not if we always send something

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go (hunk 2 above)
Comment (aduminuc) [reply backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:230]: ok , understood

**Review by aduminuc — APPROVED** [review]

**Pull request merged**